src/components/createPostModal/CreatepostModal.jsx

```diff
@@ -0,0 +1,165 @@
+        ...prevData,
+        captionError: "Caption can't be empty",
+      }));
+    }
+  };
+  return (
+    <>
+      <div className="modal modal--alert modal--post ">
+        <h5 className="modal--heading heading--4 text--center">
+          {posts.isFromEdit ? "Edit" : "Create New"} Post
+        </h5>
+        <div className="caption--container">
+          <img
+            src={allUsers?.currentUser.profilePictureUrl}
+            alt=""
+            className="avatar avatar--circle avatar--xs"
+          />
+          <div
+            className={`input--container input--${
+              postData.captionError.length ? "error" : "standard"
+            } `}
+          >
+            <label htmlFor="bio" className="input--label">
+              caption
+            </label>
+            <textarea
+              id="caption--textarea"
+              className="bio--textarea"
+              value={postData.caption}
+              placeholder="Share your thoughts here"
+              onChange={(e) => {
+                setPostData((prevData) => ({
+                  ...prevData,
+                  caption: e.target.value,
+                  captionError: "",
+                }));
+              }}
+            ></textarea>
+            <span className="input--error--message">
+              {postData.captionError}
+            </span>
+          </div>
+        </div>
+
+        {postData.imageFileError.length > 0 && (
+          <p className="input--error--message text--center">
+            {postData.imageFileError}
+          </p>
+        )}
+        {postData.imageLink !== null && postData.imageLink !== "" && (
+          <div className="post--modal--image--container">
+            <img className="img--res" src={postData.imageLink} alt="" />
+          </div>
+        )}
+
+        <div className="modal--actions">
+          <div className="file--uploader">
+            <FileUploader
+              multiple={false}
+              handleChange={handleImageUploader}
+              name="file"
+              types={fileTypes}
+              maxSize="3"
+              onSizeError={() =>
+                setPostData((prevData) => ({
+                  ...prevData,
+                  imageFileError: "Choose image less than 3MB",
+                  imageFile: null,
+                }))
+              }
+              onTypeError={() => {
+                setPostData((prevData) => ({
+                  ...prevData,
+                  imageFileError: "Choose image of type JPG,JPEG,PNG",
+                  imageFile: null,
+                }));
+              }}
+            />
+          </div>
+          <div>
+            <button
+              href="#"
+              className="btn btn--primary m-r-1"
+              onClick={closeModal}
+            >
+              Cancel
+            </button>
+
+            <button
+              href="#"
+              className="btn btn--primary "
+              onClick={clickHandler}
```

Comment:
you can give a better name to this clickHandler function. Like if it's creating a post then the name can be `createPostHandler`.